File: frontend/src/api/portal.ts
import axios from 'axios'

export interface PortalSitioItem {
  id: number
  url_dominio: string
  proveedor?: string
  fecha_pago?: string
  estatus_pago: number
  url_cpanel?: string
}

export interface PortalPagoItem {
  id: number
  concepto: string
  monto: number
  currency: string
  fecha?: string
  fecha_limite_pago?: string
  estatus: number
}

export interface PortalTicketItem {
  id: number
  titulo: string
  estado: string
  prioridad: string
  fecha_solicitud?: string
  total_notas: number
}

export interface PortalInicioResponse {
  cliente_id: number
  cliente_nombre: string
  cliente_empresa?: string
  cliente_correo?: string
  saludo: string
  total_sitios: number
  total_pendientes: number
  monto_total_pendiente: number
  total_tickets_abiertos: number
  total_hostings: number
  total_dominios: number
  doc_url: string
  sitios_recientes: PortalSitioItem[]
  pagos_pendientes_recientes: PortalPagoItem[]
  tickets_recientes: PortalTicketItem[]
}

const API_BASE = '/api/v1/portal'

export const portalApi = {
  getInicio(clienteId?: number): Promise<PortalInicioResponse> {
    const token = localStorage.getItem('access_token')
    const params = clienteId ? { cliente_id: clienteId } : {}
    return axios
      .get(`${API_BASE}/inicio`, {
        params,
        headers: {
          Authorization: `Bearer ${token}`
        }
      })
      .then((res) => res.data)
  }
}

File: backend/app/schemas/portal.py
from typing import Optional, List
from pydantic import BaseModel


class PortalSitioItem(BaseModel):
    id: int
    url_dominio: str
    proveedor: Optional[str] = None
    fecha_pago: Optional[str] = None
    estatus_pago: int = 0
    url_cpanel: Optional[str] = None


class PortalPagoItem(BaseModel):
    id: int
    concepto: str
    monto: float
    currency: str = "MXN"
    fecha: Optional[str] = None
    fecha_limite_pago: Optional[str] = None
    estatus: int = 0


class PortalTicketItem(BaseModel):
    id: int
    titulo: str
    estado: str
    prioridad: str
    fecha_solicitud: Optional[str] = None
    total_notas: int = 0


class PortalInicioResponse(BaseModel):
    cliente_id: int
    cliente_nombre: str
    cliente_empresa: Optional[str] = None
    cliente_correo: Optional[str] = None
    saludo: str
    total_sitios: int = 0
    total_pendientes: int = 0
    monto_total_pendiente: float = 0.0
    total_tickets_abiertos: int = 0
    total_hostings: int = 0
    total_dominios: int = 0
    doc_url: str = "https://conlineweb.com"
    sitios_recientes: List[PortalSitioItem] = []
    pagos_pendientes_recientes: List[PortalPagoItem] = []
    tickets_recientes: List[PortalTicketItem] = []
